# Predicting MBA Salary from grade in 10th marks

In [2]:
import pandas as pd
import numpy as np 
np.set_printoptions(precision=4)

In [3]:
mba_salary_df = pd.read_csv('MBA Salary.csv')

In [4]:
mba_salary_df.head(10)

S. No.  Percentage in Grade 10  Salary
0       1                   62.00  270000
1       2                   76.33  200000
2       3                   72.00  240000
3       4                   60.00  250000
4       5                   61.00  180000
5       6                   55.00  300000
6       7                   70.00  260000
7       8                   68.00  235000
8       9                   82.80  425000
9      10                   59.00  240000

In [5]:
mba_salary_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50 entries, 0 to 49
Data columns (total 3 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   S. No.                  50 non-null     int64  
 1   Percentage in Grade 10  50 non-null     float64
 2   Salary                  50 non-null     int64  
dtypes: float64(1), int64(2)
memory usage: 1.3 KB


In [7]:
import statsmodels.api as sm
X = sm.add_constant(mba_salary_df['Percentage in Grade 10'])

In [8]:
Y = mba_salary_df['Salary']

In [9]:
from sklearn.model_selection import train_test_split
train_X, test_X, train_Y, test_Y = train_test_split(X,
                                                   Y,
                                                   train_size = 0.8,
                                                   random_state = 100)

In [10]:
mba_salary_lm = sm.OLS(train_Y ,train_X).fit()

In [11]:
print(mba_salary_lm)

In [13]:
print(mba_salary_lm.params)

const                     30587.285652
Percentage in Grade 10     3560.587383
dtype: float64


# For every 1% increase in grade 10, the  salary of the grade 10 the salary of the MBA student increases  by 3560.587.